### ISSUE TITLE
Should we manage `nftables.service` at all? Task "Install nftables Debian systemd service unit"

### ISSUE THREAD

**Issue by stejoo** [ipr-cnrs/nftables#39]
Hi,

Thanks for your role! I have recently started using it in testing with the intention of putting it to use on several Red Hat based machines. If I happen to come across any issues I'll let you know and will try to resolve them and share a possible fix with out via PR.

I noticed task `"Install nftables Debian systemd service unit"` is being run. Modifying the original `/lib/systemd/system/nftables.service` file put in place by the `nftables` package. This happens on both EL8 (AlmaLinux 8) and the exact same should happen on Debian (11) by looking at the code, or simply: this always happens when `nft_enabled` and `nft_service_manage` are `true` (which is the default).

The unit file carried by this role seems to match the `nftables.service` file provided by Debian. Or at least it seems to closely resemble the one I compared it to (from `nftables_0.9.8-3.1_amd64.deb` (Debian 11)).  On RHEL8 the file looks a little different (compared to `nftables-0.9.3-25.el8.x86_64` (RHEL8)).
One distinct change seems to be in the `ExecReload` behaviour there:

```
-ExecReload=/sbin/nft 'flush ruleset; include "/etc/sysconfig/nftables.conf";'
+ExecReload=/usr/sbin/nft -f /etc/nftables.conf
```
The EL8 package is slightly older. But my Fedora 35 variant als has the ExecReload with the `flush ruleset` in there. So that seems a distro-specific design choice? Maybe better to keep it for RedHat based machines, to not stray from the distro too much.

Another change of importance:
```
 [Install]
-WantedBy=multi-user.target
+WantedBy=sysinit.target
```

Hmm... that does modify it's behaviour in the grand scheme of `bootup (7)` a bit compared to what's set by the package on Red Hat. Not somebody that is cause of concern, but a change that does influence it's position in the bootup order.
I noticed task "`Ensure to remove nftables systemd service from old target`" also ensures the `nftables.service` is removed from the `multi-user.target.wants`. Seems quite intentional, but I haven't figured out the motivation for it.
I am interested to learn why this is done? Perhaps the "why" could be added as comment to the task to make it more clear at the same time.

I would probably be in favor of checking if a `nftables.service` exists prior to adding a custom one. Or have such a task perform based on the distribution and it's version, for example in case Debian 10 does not supply such a file there would be need to add it ourselves. Where instead Debian 11 and EL8 do supply a `nftables.service` and we could skip adding our own.

To sum up my questions:
- Is there a need to modify/manage this file at all when it's supplied by the package?
- In case of "yes, we should". Perhaps I could add a Red Hat specific one, or perhaps set a conditional on this particular task to be skipped for systems that perhaps don't need this?
- Why remove `/etc/systemd/system/multi-user.target.wants/nftables.service` ?

Curious about what your, and possible others, thoughts are about this one.
Hope to hear from you and learn about those choices. :smiley:

**Comment by stejoo** [comment]
OK, after reading some older PRs I found: https://github.com/ipr-cnrs/nftables/pull/19

So, from what I understand: choosing `WantedBy=sysinit.target` over `WantedBy=multi-user.target` did come from the Debian .deb package I checkout out as well. And the reasoning behind it is to minimize/prevent time in a state where the network is up && configured but the firewall is not active. It sounds slightly paranoid to me, but I can understand the premise of having the firewall loaded before the network is up.

I think I understand. I'll probably add a Red Hat variant of the file, so less lines are modified.
And maybe look into making overriding it a choice for those who want to keep it as it comes shipped (I'm guessing my current client might prefer that).

**Comment by kravietz** [comment]
Generally I think we should fit the existing operating system's model of deployment as much as possible, so if on RedHat the main nft config file location is /etc/sysconfig/nftables.conf then I guess we should create ours there too. This creates some challenges because we also create a number of include files and they need to land _somewhere_ too, so this could be either /etc/sysconfig/nftables or /etc/nftables.

**Comment by stejoo** [comment]
I was able to mostly resolve my concerns and therefor this issue with PR #42. When that's merged this can be closed alongside with it imo,

**Comment by stejoo** [comment]
Thank you for merging the PRs @kravietz 
Closing this one. :smile:
